cpp/ql/src/semmle/code/cpp/Variable.qll

```diff
@@ -176,7 +189,21 @@ class Variable extends Declaration, @variable {
 }
 
 /**
- * A particular declaration or definition of a C/C++ variable.
+ * A particular declaration or definition of a C/C++ variable. For example in
+ * the following code there are five variable declaration entries - two for
+ * `a`, and one each for `b`, `c` and `d`:
+ * ```
+ * extern int a;
+ * int a;
+ *
+ * void myFunction(int b) {
+ *   int c;
+ * }
+ *
+ * namespace N {
+ *   int d;
+ * }
+ * ```
```

Comment:
Perhaps `namespace N { extern int d; }`?

Reply:
Do you just want to see more examples where the `VariableDeclarationEntry` is not a definition?

Reply:
Yes, I think that's a good idea -- to mix up `extern` and non-`extern` declarations (given that you've explained how `VariableDeclarationEntry`s actually work).

Reply:
Done.